### PR TITLE
oldClassConstructors: handle abstract classes

### PR DESCRIPTION
**Pull request by macbre**
```php
abstract class TheThing {
	public function TheThing() {
	}
}
```

Improves #18

### CODE REVIEW

File: classes/tests/critical.php [hunk 1]
```diff
@@ -144,10 +144,8 @@ public function _oldClassConstructors($line) {
 		}
 
 		// find the start of PHP class declaration
-		if (strpos($line, 'class') === 0) {
-			if (preg_match('#class (\w+)#', $line, $matches)) {
-				$lastClassName = $matches[1];
-			}
+		if (preg_match('#^(abstract\s+)?class (\w+)#', $line, $matches)) {
```
Comment (Alexia) [inline classes/tests/critical.php]: Minor regression, that ^ beginning of line character will not match code that has white space at the beginning.(Poor formatting and such.)

Comment (macbre) [inline classes/tests/critical.php]: Good point, improved in c89242c

**Comment by Alexia** [comment]
Thanks!  Works on my end too.

**Comment by macbre** [comment]
Great, thanks!

**Pull request merged**